djgeojson/views.py

```diff
@@ -88,7 +93,19 @@ def get_queryset(self):
         """
         Inspired by Glen Roberton's django-geojson-tiles view
         """
-        self.z, self.x, self.y = map(int, self.args[:3])
+        try:
+            self.z, self.x, self.y = map(int, self.args[:3])
+        except AttributeError:
+            # let's try to get these as kwargs
+
+            try:
+
+                self.z = int(self.kwargs['z'])
+                self.x = int(self.kwargs['x'])
+                self.y = int(self.kwargs['y'])
+            except (ValueError, TypeError, KeyError):
+                return HttpResponseBadRequest(u"View parameters could not be processed.")
```

Comment:
By the docs you cannot raise a HttpResponseBadRequest.

We can just let the exceptions blow up and catch them while trying to fetch the queryset in the GeoJSONMixin.